inputremapper/injection/consumers/mapping_parser.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# input-remapper - GUI for device specific keyboard mappings
+# Copyright (C) 2022 sezanzeb <proxima@sezanzeb.de>
+#
+# This file is part of input-remapper.
+#
+# input-remapper is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# input-remapper is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with input-remapper.  If not, see <https://www.gnu.org/licenses/>.
+"""functions to assemble the mapping handlers"""
+
+from typing import Dict, List, Type
+
+from evdev.ecodes import EV_KEY, EV_ABS, EV_REL
+
+from inputremapper.logger import logger
+from inputremapper.key import Key
+from inputremapper.injection.consumers.mapping_handler import (
+    MappingHandler,
+    CombinationHandler,
+    ContextProtocol,
+    HierarchyHandler,
+    AbsToBtnHandler,
+    RelToBtnHandler,
+)
+from inputremapper.mapping import Mapping
+
+MappingHandlers = Dict[Key, List[MappingHandler]]
+
+mapping_handler_classes: Dict[str, Type[MappingHandler]] = {
+    # all available mapping_handlers
+    "combination": CombinationHandler,
+}
+
+
+def parse_mapping(mapping: Mapping, context: ContextProtocol) -> MappingHandlers:
+    """create dict with a list of MappingHandlers for each key
+
+    Key is of len 1
+    """
+
+    # create a handler for each mapping
+    combination_handlers = {}
+    normal_handlers = {}
+
+    for key, sub_mapping in mapping:
```

Comment:
And maybe we should change the iterator of the `Mapping` class to yield `key, symbol, target`

Reply:
let us rename the mapping to preset, as you suggested.
And then I think the preset should yield a mapping object. Which would be a Dataclass
I would like to do something like:
```
for mapping in preset:
   handlers[mapping.key] = MappingHandler(mapping)
```
A mapping can be so much more than just some keys to other keys or macros. For each abs axis we can individually specify the gain, expo, deadzone, ...
We also can map multiple buttons to one ABS axis by just specifying  different trigger_points.
currently I have a lot of methods in `mapping_parser`  which construct different dicts for the mappings, but all that should happen in the preset class an ultimately represented in the save file

Reply:
Sounds intuitive, yes

Reply:
I just renamed the Mapping to Preset, and went over all the docstrings, to untangle things. There might be some wrong docsrtings, which actually refer to a mapping but the word `mapping` got replaced with `preset`. I am not sure if I caught all of those.

Also there is a file `presets.py` which contains methods for managing presets. I will go over them and see if I can move them to the `Preset` class, make them member- or class-functions